gateway-service/src/main/kotlin/zhi/yest/communityintersection/gatewayservice/security/config/SecurityConfig.kt

```diff
@@ -15,32 +18,47 @@ import org.springframework.security.oauth2.client.web.server.ServerOAuth2Authori
 import org.springframework.security.oauth2.core.user.DefaultOAuth2User
 import org.springframework.security.oauth2.core.user.OAuth2UserAuthority
 import org.springframework.security.web.server.SecurityWebFilterChain
-import org.springframework.web.cors.CorsConfiguration
-import org.springframework.web.cors.reactive.CorsWebFilter
-import org.springframework.web.cors.reactive.UrlBasedCorsConfigurationSource
 import org.springframework.web.reactive.function.client.WebClient
 import org.springframework.web.reactive.function.client.bodyToFlux
+import reactor.core.publisher.Mono
 import reactor.core.publisher.toMono
-import zhi.yest.vk.friendfinder.config.security.dto.VkResponse
-import zhi.yest.vk.friendfinder.config.security.dto.VkUserInfo
-import java.util.Arrays
+import zhi.yest.communityintersection.gatewayservice.dto.VkResponse
+import zhi.yest.communityintersection.gatewayservice.dto.VkUserInfo
+import zhi.yest.communityintersection.gatewayservice.security.VkCodeTokenResponseClient
 
 @Configuration
 @EnableWebFluxSecurity
 class SecurityConfig {
     @Bean
     fun configure(http: ServerHttpSecurity,
                   resolver: ServerOAuth2AuthorizationRequestResolver,
-                  authManager: ReactiveAuthenticationManager): SecurityWebFilterChain {
+                  authManager: ReactiveAuthenticationManager,
+                  filterFactory: TokenRelayGatewayFilterFactory): SecurityWebFilterChain {
+        // @formatter:off
         return http
                 .authorizeExchange()
-                .anyExchange().authenticated()
+                    .anyExchange().authenticated()
                 .and()
-                .oauth2Login().authenticationManager(authManager)
+                    .oauth2Login().authenticationManager(authManager)
                 .and()
-                .addFilterAt(corsFilter(), SecurityWebFiltersOrder.CORS)
                 .addFilterAt(OAuth2AuthorizationRequestRedirectWebFilter(resolver), SecurityWebFiltersOrder.FIRST)
+                //TODO: enable CSRF protection
+                .csrf().disable()
                 .build()
+        // @formatter:on
+    }
+
+    // Spring Cloud Gateway for some reason removes the SESSION cookie via set-cookie header
```

Comment:
Why do we need a session cookie?

Reply:
The `access_token` is stored on the back end and is never sent to UI application. Once a user is authorized, `gateway-service` creates a session and assigns a cookie for the UI client the same way it was done before.

The problem is that for some reason once the client makes a call to any of the resource services, the `gateway-service` sends a `set-cookie: SESSION=;` header to the client and effectively removes the cookie from browser.